packages/access-api/test/upload-api-proxy.test.js

```diff
@@ -0,0 +1,115 @@
+import assert, { AssertionError } from 'assert'
+import * as Store from '@web3-storage/capabilities/store'
+import * as Upload from '@web3-storage/capabilities/upload'
+import * as dagUcanDid from '@ipld/dag-ucan/did'
+import { context } from './helpers/context.js'
+import * as ucanto from '@ucanto/core'
+// eslint-disable-next-line no-unused-vars
+import * as iucanto from '@ucanto/interface'
+import { isUploadApiStack } from '../src/service/upload-api-proxy.js'
+
+describe('parserCapabilities', function () {
```

Comment:
These tests seem redundant to me, they basically test that `parserAbilities` works by comparing to the list of abilities expected. It seems to me that just using a set of `expectedCans` in the next tests would be good enough and more practical.

Only thing we can catch with the whole `parserAbilities` dance is new capabilities added, yet there's no guarantee that `parserAbilities` will catch those. If do want to go extra mile and catch those cases, I would suggest using `can` field of capability parsers (they should have it although it's not part of interface), in other words you could derive expected cans like:

```ts
Object.values(Upload).map(cap => cap.can || '').filter(can => !can.includes('|'))
```

Reply:
That approach kind of works, but I found it hard to make into a well-typed reusable function (replace `Upload` with a parameter of what type?).

I removed the tests though, which aren't adding much.

Reply:
https://github.com/web3-storage/w3protocol/pull/334/commits/51794b0a3ffdb6a544f2f4b66c431ecb5aec5126

Reply:
> That approach kind of works, but I found it hard to make into a well-typed reusable function (replace Upload with a parameter of what type?).

Yeah because `can` isn't in types.

I'm mostly concerned that some assumbtions `parserAbilities` makes may not stand long term. It's fine though, we can deal with that when it happens.